File: zig/darts/darts.zig
const std = @import("std");

pub const Coordinate = struct {
    x: f32,
    y: f32,

    pub fn init(x_coord: f32, y_coord: f32) Coordinate {
        return Coordinate{ .x = x_coord, .y = y_coord };
    }

    pub fn score(self: Coordinate) isize {
        const distance = std.math.hypot(f32, self.x, self.y);
        if (distance <= 1.0) return 10;
        if (distance <= 5.0) return 5;
        if (distance <= 10.0) return 1;
        return 0;
    }
};

File: python/darts/darts.py
from math import hypot, ceil


def score(x, y):
    match ceil(hypot(x, y)):
        case 0 | 1: return 10
        case 2 | 3 | 4 | 5: return 5
        case 6 | 7 | 8 | 9 | 10: return 1
        case _: return 0
